DeviceAdministration/Infrastructure/Repository/DeviceTelemetryRepository.cs

```diff
@@ -263,45 +263,77 @@ public DeviceTelemetryRepository(IConfigurationProvider configProvider)
+                    DateTime date;
+                    if (strdict.TryGetValue("eventenqueuedutctime", out str) &&
+                        DateTime.TryParse(
+                            str, 
                             CultureInfo.InvariantCulture,
-                            out number))
+                            DateTimeStyles.AllowWhiteSpaces,
+                            out date))
                     {
-                        model.ExternalTemperature = number;
+                        model.Timestamp = date;
                     }
 
-                    if (strdict.TryGetValue("humidity", out str) &&
-                        double.TryParse(
-                            str,
-                            NumberStyles.Float,
-                            CultureInfo.InvariantCulture,
-                            out number))
+                    IEnumerable<DeviceTelemetryFieldModel> fields;
+
+                    if (telemetryFields != null && telemetryFields.Count > 0)
                     {
-                        model.Humidity = number;
+                        fields = telemetryFields;
                     }
-
-                    if (strdict.TryGetValue("temperature", out str) &&
-                        double.TryParse(
-                            str,
-                            NumberStyles.Float,
-                            CultureInfo.InvariantCulture,
-                            out number))
+                    else
                     {
-                        model.Temperature = number;
+                        List<string> reservedColumns = new List<string>
+                        {
+                            "DeviceId",
+                            "EventEnqueuedUtcTime",
+                            "EventProcessedUtcTime",
+                            "IoTHub",
+                            "PartitionId"
+                        };
+
+                        fields = strdict.Keys
+                            .Where((key) => !reservedColumns.Contains(key))
+                            .Select((name) => new DeviceTelemetryFieldModel
+                            {
+                                Name = name,
+                                Type = "double"
+                            });
                     }
 
-                    DateTime date;
-                    if (strdict.TryGetValue("eventenqueuedutctime", out str) &&
-                        DateTime.TryParse(
-                            str, 
-                            CultureInfo.InvariantCulture,
-                            DateTimeStyles.AllowWhiteSpaces,
-                            out date))
+                    foreach (var field in fields)
                     {
-                        model.Timestamp = date;
+                        if (strdict.TryGetValue(field.Name, out str))
+                        {
+                            switch (field.Type)
+                            {
+                                case "int":
```

Comment:
Looking at the types available there, here's what I'll do:

Parsed as integer:

* int
* int16
* int32
* int64
* sbyte
* byte

Parsed as float
* double
* decimal
* single

None of the other types appear to map to a reasonable numeric value, so I'm going to ignore them for this